crates/dapf/src/main.rs

```diff
@@ -116,6 +136,82 @@ impl CliVdafConfig {
     }
 }
 
+#[derive(Clone, Debug)]
+struct CliDapQueryConfig(DapQueryConfig);
+
+impl fmt::Display for CliDapQueryConfig {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        <DapQueryConfig as fmt::Display>::fmt(&self.0, f)
+    }
+}
+
+impl From<CliDapQueryConfig> for DapQueryConfig {
+    fn from(CliDapQueryConfig(val): CliDapQueryConfig) -> Self {
+        val
+    }
+}
+
+impl From<DapQueryConfig> for CliDapQueryConfig {
+    fn from(value: DapQueryConfig) -> Self {
+        Self(value)
+    }
+}
+
+impl FromStr for CliDapQueryConfig {
+    type Err = String;
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        if s == "time-interval" {
+            Ok(Self(DapQueryConfig::TimeInterval))
+        } else if let Some(size) = s.strip_prefix("fixed-size") {
+            Ok(Self(DapQueryConfig::FixedSize {
+                max_batch_size: if let Some(size) = size.strip_prefix("-") {
+                    Some(
+                        size.parse()
+                            .map_err(|e| format!("{s} is an invalid query config: {e:?}"))?,
+                    )
+                } else if size.is_empty() {
+                    None
+                } else {
+                    return Err(format!("{size} is an invalid fixed size max batch size"));
+                },
+            }))
+        } else {
+            Err(format!("{s} is an invalid query config"))
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+struct CliTaskId(TaskId);
```

Comment:
Is Cli short for "command line interface" or is it short for "Client"? Why do we need a wrapper for task IDs here?

Reply:
We need a wrapper to TaskId to impl `FromStr` for it. Previously this problem was solved by using `value_parser` from clap, but by using this new type we can also ask the user interactively. See `request_from_user`

Reply:
Cli stands for command line interface. I intend to move all these `Cli*` types to their own module which will make this less confusing in the future